clang/lib/CIR/Lowering/DirectToLLVM/LowerToLLVM.cpp

```diff
@@ -2452,58 +2458,70 @@ mlir::LogicalResult CIRToLLVMBinOpLowering::matchAndRewrite(
 
   switch (op.getKind()) {
   case cir::BinOpKind::Add:
-    if (mlir::isa<cir::IntType, mlir::IntegerType>(type))
+    if (mlir::isa<cir::IntType, mlir::IntegerType>(type)) {
```

Comment:
I'm puzzled on why did you add a `isIntTypeUnSigned` but didn't update the other places that do the exact same check. 

We discussed this briefly over VC: you might wanna look into how `adaptor`s work in lowering, it's super weird we are checking both CIR and LLVM types, can you elaborate on why only checking the LLVM type via adaptor isn't enough? I know this was before your PR, but I'd prefer to see this fixed so we stop the bad pattern.

Reply:
I see. I didn't know LLVM types here meant `mlir::IntegerType` in this context. Yes, I did try to remove the check for `mlir::IntegerType`, but got a test failure as it seems one of our tests for cir::BinOpKind::Add is expecting `mlir::IntegerType` in this lowering code . I'll take investigate more to see if it is necessary. In general, I think we ought to just expect cir::IntType in this context.

Reply:
> Yes, I did try to remove the check for mlir::IntegerType

If you can do it on top of cir::IntType, even better, but I suggested quite the opposite: eliminate the cir::IntType check and use mlir::IntegerType on the type you can obtain from the adaptor. The problem with cir::IntType is that some of the operands might already been lowered, with the adaptor you guarantee LLVM only.

Reply:
Got it. So Int type check, we can just use something like `mlir::isa<mlir::IntegerType>(adaptor.getRhs().getType())`
or even `mlir::isa<mlir::IntegerType>(llvmTy)`, 
but maybe for unsigned and signed detection, just do it for cir::IntType as llvm type is signless?

Reply:
> mlir::isa<mlir::IntegerType>(adaptor.getRhs().getType())

Yep

> unsigned and signed detection

If we need to detect signed/unsigned then CIR types are the way, but I forgot about those, then probably what would be best is to only rely on CIR types (and not look at LLVM at all), I don't if that's possible or I don't know the root cause behind checking both types. It's worth some investigation, but thinking more about this: maybe better to tackle it in a separate PR later on? Let's perhaps just add/sub on top of the existing practice and you can investigate cleaning up this multiple dialect type checking later.

Reply:
Sure, I'll just keep the simple refactoring as it stands now for this PR, and investigate test failure when I changed it into `mlir::isa<cir::IntType>`, and to see why we have mixed MLIR integer type and cir IntType here.

Reply:
Sounds great